### PR TITLE
created angle_convert variables  and changed theta and phi to angle1 …

### PR DESCRIPTION
**Pull request by henry-lignux**
good

### CODE REVIEW

File: spherical-gear.h [hunk 1]
```diff
@@ -1,34 +1,38 @@
 #include <Servo.h> // for the motor
 #include <cmath> // for the math
 #define TIME_ITERATION 10 // used for servo delay
+#define ANGLE_CONVERT_1
```
Comment (henry-lignux) [inline spherical-gear.h]: This is invalid code. define ANGLE_CONVERT_1 and 2 as 1.0 as default

File: spherical-gear.h [hunk 2]
```diff
@@ -1,34 +1,38 @@
...
+    double angle2;
 	public:
 		SphericalGear(int pin1, int pin2) {
       // initializing the motors
       servo1.attach(pin1);
       servo2.attach(pin2);
       // initializing the angles
-      theta = 0;
-      phi = 0;
+      angle1 = 0;
```
Comment (henry-lignux) [inline spherical-gear.h:20]: make sure to comment that this is the angle of the motor not the sphere.

File: spherical-gear.h [hunk 3]
```diff
@@ -1,34 +1,38 @@
...
     void move(double delta_theta, double delta_phi, double angular_speed) {
       // radius is assumed to be 1
+      // convert from spherical angle to motor angle
       // measured in degrees; from mr. nims's equation
-      double total_angle_change = acos( sin(theta)*sin(theta+delta_theta)*cos(phi-delta_phi) + cos(theta)*cos(theta+delta_theta) );
+      double total_angle_change = acos( sin((angle1*ANGLE_CONVERT_1))*sin((angle1*ANGLE_CONVERT_1)+delta_theta)*cos((angle2*ANGLE_CONVERT_2)-delta_phi)
+       + cos((angle1*ANGLE_CONVERT_1))*cos((angle1*ANGLE_CONVERT_1)+delta_theta) );
       double time = total_angle_change / angular_speed; // in miliseconds
       double theta_speed = delta_theta / time;
       double phi_speed = delta_phi / time;
```
Comment (henry-lignux) [inline spherical-gear.h]: change theta_speed  and phi_speed to angle1_speed and angle2_speed. And make sure to divide by angle convert

File: spherical-gear.h [hunk 4]
```diff
@@ -1,34 +1,38 @@
...
       for(int t = 0; t <= time; t+=TIME_ITERATION) { 
 	      // iterates over TIME_ITERATION miliseconds 
-	      servo1.write(theta); // moves motor to theta
-        servo2.write(phi); // moves motor to phi
-        theta += theta_speed*TIME_ITERATION;
-        phi += phi_speed*TIME_ITERATION;
+	      servo1.write(angle1); // moves motor to theta
+        servo2.write(angle2); // moves motor to phi
+        angle1 += theta_speed*TIME_ITERATION;
+        angle2 += phi_speed*TIME_ITERATION;
```
Comment (henry-lignux) [inline spherical-gear.h]: make sure to change theta_speed and phi_speed here too.

**Review by henry-lignux — COMMENTED** [review]
change this

**Review by sparshb13 — APPROVED** [review]

**Review by sparshb13 — APPROVED** [review]

**Pull request merged**